src/util/performance.js

```diff
@@ -117,67 +96,25 @@ export const PerformanceUtils = {
-        // multiples of 1000/60ms (+-2ms).
-        //
-        // The differences between the timestamps vary a lot more in Safari.
-        // It's not uncommon to see a 24ms difference followedd by a 8ms difference.
-        // I'm not sure, but I think these might not be dropped frames (due to multiple
-        // buffering?).
-        //
-        // For Safari, I think comparing the number of expected frames with the number of actual
-        // frames is a more accurate way of measuring dropped frames than comparing
-        // individual frame time differences to a target time. In Firefox and Chrome
-        // both approaches produce the same result most of the time.
-        let droppedFrames = 0;
-        let totalFrameTimeSum = 0;
-        let totalFrames = 0;
-        metrics.jank = 0;
-
-        for (const frameTimes of frameSequences) {
-            if (!frameTimes.length) continue;
-            const frameTimeSum = frameTimes.reduce((prev, curr) => prev + curr, 0);
-            const expectedFrames = Math.max(1, Math.round(frameTimeSum / frameTimeTarget));
-            droppedFrames += expectedFrames - frameTimes.length;
-            totalFrameTimeSum += frameTimeSum;
-            totalFrames += frameTimes.length;
-
-            // Jank is a change in the frame rate.
-            // Count the number of times a frame has a worse rate than the previous frame.
-            // A consistent rate does not increase jank even if it is continuosly dropping frames.
-            // A one-off frame does not increase jank even if it is really long.
-            //
-            // This is not that accurate in Safari because the differences between animation frame
-            // times is not as close to a multiple of 1000/60ms.
-            const roundedTimes = frameTimes.map(frameTime => Math.max(1, Math.round(frameTime / frameTimeTarget)));
-            for (let n = 0; n < roundedTimes.length - 1; n++) {
-                if (roundedTimes[n + 1] > roundedTimes[n]) {
-                    metrics.jank++;
-                }
-            }
-        }
-        const avgFrameTime = totalFrameTimeSum / totalFrames / 1000;
-        metrics.fps = 1 / avgFrameTime;
-        metrics.droppedFrames = droppedFrames;
-        metrics.percentDroppedFrames = (droppedFrames / (totalFrames + droppedFrames)) * 100;
-
-        metrics.cpuFrameBudgetExceeded = 0;
-        const renderFrames = performance.getEntriesByName('render');
-        for (const renderFrame of renderFrames) {
-            metrics.cpuFrameBudgetExceeded += Math.max(0, renderFrame.duration - CPU_FRAME_BUDGET);
-        }
-
         metrics.placementTime = placementTime;
 
         return metrics;
     },
 
     getWorkerPerformanceMetrics(): { timeOrigin: string, measures: Array<PerformanceEntry> } {
+        const entries = performance.getEntries().map(entry => {
+            const result = JSON.parse(JSON.stringify(entry));
```

Comment:
Nit: there's https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/toJSON

Reply:
fixed in [17a61bb](https://github.com/mapbox/mapbox-gl-js/pull/11828/commits/17a61bbec9362d7120e3e53c1ca04ca2c511ac68)